### DISCUSSION TITLE
Installation options - Architecture to install: Default / x86 / x64

### DISCUSSION THREAD

**Discussion by archibaldodlc** [Devolutions/UniGetUI#2897] [General]
Hi, everyone!

 I would like to start saying that recently I found this app and it's amazing. I have one question refers to the Arquitecture options for install applications. The app has three architecture options: Default, x86 and x66. The last ones is very clear, but "Default" which means? It will install the x86 or x64 version?

I would like to install the x64 version for all my apps, unless they don't exist or not to be available, in which case I would like to install  the x86 version.

I edited the bundle file to set the architecture to x64 like this:

```
{
      "Id": "Flow-Launcher.Flow-Launcher",
      "Name": "Flow Launcher",
      "Version": "1.19.3",
      "Source": "winget",
      "ManagerName": "Winget",
      "InstallationOptions": {
        "SkipHashCheck": false,
        "InteractiveInstallation": false,
        "RunAsAdministrator": false,
        **"Architecture": "x64",**
        "InstallationScope": "machine",
        "CustomParameters": [],
        "PreRelease": false,
        "CustomInstallLocation": "",
        "Version": ""
      },
      "Updates": {
        "UpdatesIgnored": false,
        "IgnoredVersion": ""
      }
    },
```

but for some packeges I got that cannot find and installer.

It is possible to achieve that in UniGetUi?

Thanks in advance

**Comment by marticliment** [comment]
No, but the architecture is set to Default (an empty string) the package manager will choose the most suitable installer for that package. (x64 on a x64 system, x86 if there is no x64 version, arm64 on arm64 system, etc.)

**Comment by StHofmann** [comment]
First of all, thank you very much for this excellent tool that makes life that much easier!

For the example of the Java runtime environment, it could be beneficial to support the parallel installation of more than one architecture. In my case, I have a Java library in some legacy application (with no update available) that fails with x64, but works fine with x86. But currently, UniGetUI can only install and update only one architecture, leaving the other for manual installation and update. 

When installing a package for one architecture (x86) and once again for another architecture (x64), the package appears twice in the list of installed packages, but UniGetUI changes the architecture setting for both packages to the architecture of the last install.

So how about allowing a package to be installed more than once with different architectures and check/reject the installation with an architecture already installed?

**Reply by StHofmann** [reply]
PS: UniGetUI adds a manually installed package of another architecture, but discards its original architecture setting. So you have twice the package in the list with identical architecture.

**Reply by marticliment** [reply]
I will see what I can do

**Comment by StHofmann** [comment]
Just received the upgrade to version 3.3.0 – and instantly missed the one-click option to update the selected packages. The new pull-down menu causes some extra clicks…

Another point: I received the upgrade on one machine. But after clicking the "Check for updates" button on another machine to also get the upgrade there, I was told that version 3.2.0 still is the most recent one. How come?

**Reply by marticliment** [reply]
> Another point: I received the upgrade on one machine. But after clicking the "Check for updates" button on another machine to also get the upgrade there, I was told that version 3.2.0 still is the most recent one. How come?

I have disabled auto updates. So you may just have picked the point where I disabled them.
You can still update manually

**Reply by StHofmann** [reply]
Not sure I made the point clear, but manual update does not work for that other machine (see the screenshot).

<img width="1004" height="425" alt="image" src="https://github.com/user-attachments/assets/73a492a0-eb34-419e-a2dc-90323ba75d26" />

It still stays at the old version.

**Reply by marticliment** [reply]
I have also disabled updates to 3.3.0 stable due to some crashes

**Reply by StHofmann** [reply]
Got it, thanks for clarification.

**Reply by StHofmann** [reply]
> Just received the upgrade to version 3.3.0 – and instantly missed the one-click option to update the selected packages. The new pull-down menu causes some extra clicks…

Thanks for offering the one-click update in 3.3.1 again.
